### PR TITLE
Make driver functionality to work and adopt latest csi-lib-iscsi and csi spec 1.5.0

### PR DESCRIPTION
**Pull request by humblec**
Considering csi-lib-iscsi has gone through some refactoring, it is
good to consume latest commit and also to use latest
csi spec ie 1.5.0.

This commit update the dependency of below projects:

- github.com/container-storage-interface/spec v1.5.0
- github.com/kubernetes-csi/csi-lib-iscsi v0.0.0-20211110090527-5c802c48a124
- github.com/kubernetes-csi/csi-lib-utils v0.10.0
- google.golang.org/grpc v1.38.0

The ISCSI driver functionality has been tested with this PR and it looks to be working as expected :+1: 
Please refer # https://github.com/kubernetes-csi/csi-driver-iscsi/pull/70#issuecomment-984742171

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by andyzhangx** [comment]
can you run `make && make sanity-test` on your local env? not sure why github action does not work, let me add a CI test pipeline to run sanity test first.

**Comment by humblec** [comment]
> can you run `make && make sanity-test` on your local env? not sure why github action does not work, let me add a CI test pipeline to run sanity test first.

@andyzhangx indeed it was ran and I wanted to put the fixes as followup commits https://github.com/kubernetes-csi/csi-driver-iscsi/pull/70/commits/47f010f122cd1d8e585a8b8c5d116b15c5d50f10 here to marked in WIP, once the verification is done I will remove the WIP.  :+1:

**Comment by andyzhangx** [comment]
aha, I see, the sanity test is already running there, need to rename to `Sanity tests`, here is the PR: https://github.com/kubernetes-csi/csi-driver-iscsi/pull/71

**Comment by humblec** [comment]
```


root@hchiramm csi-driver-iscsi]# kubectl get csidriver
NAME               ATTACHREQUIRED   PODINFOONMOUNT   STORAGECAPACITY   TOKENREQUESTS   REQUIRESREPUBLISH   MODES        AGE
iscsi.csi.k8s.io   false            false            false             <unset>         false               Persistent   12m
[root@hchiramm csi-driver-iscsi]# 

I1202 13:05:02.878282       8 utils.go:64] GRPC request: {}
I1202 13:05:02.879757       8 identityserver.go:32] Using default GetPluginInfo
I1202 13:05:02.879762       8 utils.go:69] GRPC response: {"name":"iscsi.csi.k8s.io","vendor_version":"1.0.0"}
I1202 13:05:02.999608       8 utils.go:63] GRPC call: /csi.v1.Identity/GetPluginInfo
I1202 13:05:02.999632       8 utils.go:64] GRPC request: {}


[root@hchiramm static-pv-name]# kubectl get pods
NAME                   READY   STATUS    RESTARTS   AGE
csi-iscsi-node-ndchq   3/3     Running   0          10m
[root@hchiramm static-pv-name]# 

[root@hchiramm csi-driver-iscsi]# kubectl get pvc
NAME        STATUS   VOLUME           CAPACITY   ACCESS MODES   STORAGECLASS   AGE
iscsi-pvc   Bound    static-pv-name   1Gi        RWO                           12m

----------------

PV :

Message:         
Source:
    Type:              CSI (a Container Storage Interface (CSI) volume source)
    Driver:            iscsi.csi.k8s.io
    FSType:            
    VolumeHandle:      hum-iscsi-share
    ReadOnly:          false
    VolumeAttributes:      discoveryCHAPAuth=false
                           iqn=iqn.2015-06.com.example.test:target1
                           iscsiInterface=default
                           lun=1
                           portals=[]
                           targetPortal=127.0.0.1:3260
Events:                <none>


[root@hchiramm csi-driver-iscsi]# kubectl create -f pod.yaml
pod/task-pv-pod created

[root@hchiramm csi-driver-iscsi]# ls -l /dev/disk/by-path/* |grep iscsi
lrwxrwxrwx 1 root root  9 Dec  2 18:57 /dev/disk/by-path/ip-127.0.0.1:3260-iscsi-iqn.2015-06.com.example.test:target1-lun-1 -> ../../sda

[root@hchiramm csi-driver-iscsi]# lsblk |grep mount
sda                               8:0    0     2G  0 disk /var/lib/kubelet/pods/d8718ae2-58b0-417f-99ec-59d5389b5889/volumes/kubernetes.io~csi/static-pv-name/mount
[root@hchiramm csi-driver-iscsi]# 
[root@hchiramm csi-driver-iscsi]# kubectl exec -ti task-pv-pod -- bash
root@task-pv-pod:/# 
root@task-pv-pod:/# cd /usr/share/nginx/html/
root@task-pv-pod:/usr/share/nginx/html# touch iscsi
root@task-pv-pod:/usr/share/nginx/html# echo "iscsi" >isci
root@task-pv-pod:/usr/share/nginx/html# ls
isci  iscsi  lost+found  test
root@task-pv-pod:/usr/share/nginx/html# cat isci 
iscsi
root@task-pv-pod:/usr/share/nginx/html# exit
exit
[root@hchiramm csi-driver-iscsi]# kubectl get pods
NAME                   READY   STATUS    RESTARTS   AGE
csi-iscsi-node-ndchq   3/3     Running   0          11m
task-pv-pod            1/1     Running   0          70s

[root@hchiramm csi-driver-iscsi]# kubectl delete pod task-pv-pod
pod "task-pv-pod" deleted
[root@hchiramm csi-driver-iscsi]# 

[root@hchiramm csi-driver-iscsi]# ls -l /dev/disk/by-path/*

[root@hchiramm csi-driver-iscsi]# 
```

**Review by j-griffith — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -12,13 +12,11 @@
...
-FROM centos:7.4.1708
+FROM ubuntu
 
 # Copy iscsiplugin.sh
 COPY iscsiplugin.sh /iscsiplugin.sh
 # Copy iscsiplugin from build _output directory
-COPY bin/iscsiplugin /iscsiplugin
+COPY ./bin/iscsiplugin /iscsiplugin
 
-RUN yum -y install iscsi-initiator-utils e2fsprogs xfsprogs && yum clean all
```
Comment (andyzhangx) [inline Dockerfile:22]: these packages are all not needed? if so, base image could be a distroless image, that could save size.

File: deploy/csi-iscsi-node.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: configmap-csi-iscsiadm
+data:
+  iscsiadm: |
+    #!/bin/sh
+    if [ -x /host/sbin/iscsiadm ]; then
+      chroot /host /sbin/iscsiadm "$@"
```
Comment (andyzhangx) [inline deploy/csi-iscsi-node.yaml:143]: what's this config map for? maybe we should move this script to the iscsi driver initContainer or inside driver?

Comment (humblec) [reply deploy/csi-iscsi-node.yaml:143]: actually it is in the driver pod as a volume.

File: Dockerfile (hunk 1 above)
Comment (humblec) [reply Dockerfile:22]: yeah, with above base image in place we get the prerequisites.. not sure distroless base could miss some prereqisites, but I can cross check of using that as well.

**Pull request merged**